x-pack/legacy/plugins/monitoring/server/lib/logs/detect_reason_from_exception.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function detectReasonFromException(exception) {
+  const reason = {};
```

Comment:
I think we should have a default here, since if the bottom condition does not pass and somebody checks falsey via `!reason.correctIndexName` it will give a false positive.

Also, checking truthy condition will always fail eg `if (reason.correctIndexName)`, since it'll either be `undefined` or `false`
```suggestion
  const reason = { correctIndexName: true };
```

Reply:
Fixed in : https://github.com/elastic/kibana/pull/48929/commits/cf4e0e394aa4b4f6514c15ed04f888a132851096